vars/runATH.groovy

```diff
@@ -0,0 +1,205 @@
+            isLocalATH = athUrl.startsWith("file://")
+            athRevision = metadata.athRevision ?: athRevision
+
+            // Allow override of jenkins version from metadata file
+            jenkins = metadata.jenkins ?: jenkins
+            isVersionNumber = (jenkins =~ /^\d+([.]\d+)*$/).matches()
+
+            if (!isLocalATH) {
+                echo 'Checking connectivity to ATH sources…'
+                sh "git ls-remote --exit-code -h ${athUrl}"
+            }
+            if (jenkins == "latest") {
+                jenkinsURl = mirror + "war/latest/jenkins.war"
+            } else if (jenkins == "latest-rc") {
+                jenkinsURl = mirror + "/war-rc/latest/jenkins.war"
+            } else if (jenkins == "lts") {
+                jenkinsURl = mirror + "war-stable/latest/jenkins.war"
+            } else if (jenkins == "lts-rc") {
+                jenkinsURl = mirror + "war-stable-rc/latest/jenkins.war"
+            }
+
+            if (!isVersionNumber) {
+                echo 'Checking whether Jenkins WAR is available…'
+                sh "curl -ILf ${jenkinsURl}"
+            }
+            // Validation ended
+
+            // ATH
+            if (isLocalATH) { // Deal with already existing ATH sources
+                athSourcesFolder = athUrl - "file://"
+            } else {
+                dir(athSourcesFolder) {
+                    checkout changelog: true, poll: false, scm: [$class           : 'GitSCM', branches: [[name: athRevision]],
+                                                                 userRemoteConfigs: [[url: athUrl]]]
+                }
+            }
+            dir(athSourcesFolder) {
+                // We may need to run things in parallel later, so avoid several checkouts
+                stash name: "athSources"
+            }
+
+            // Jenkins war
+            if (isVersionNumber) {
+                def downloadCommand = "mvn dependency:copy -Dartifact=org.jenkins-ci.main:jenkins-war:${jenkins}:war -DoutputDirectory=. -Dmdep.stripVersion=true"
+                dir("deps") {
+                    if (infra.isRunningOnJenkinsInfra()) {
+                        def settingsXml = "${pwd tmp: true}/settings-azure.xml"
+                        writeFile file: settingsXml, text: libraryResource('settings-azure.xml')
+                        downloadCommand = downloadCommand + " -s ${settingsXml}"
+                    }
+                    withEnv(env) {
+                        sh downloadCommand
+                    }
+                    sh "cp jenkins-war.war jenkins.war"
+                    stash includes: 'jenkins.war', name: 'jenkinsWar'
+                }
+            } else {
+                sh("curl -o jenkins.war -L ${jenkinsURl}")
+                stash includes: '*.war', name: 'jenkinsWar'
+            }
+
+        }
+    })
+
+    ensureInNode(env, env.RUN_ATH_DOCKER_NODE ?: 'docker && highmem', {
+        if (skipExecution) {
+            return
```

Comment:
Is this a code style thing? this is really not something you normally do in groovy.